### PR TITLE
refactor(consent-portal): make configuration vendor-neutral and consistent

### LINKED ISSUE #464: Make consent-portal configuration vendor-neutral and consistent
## Current Limitation

The **consent-portal** carries vendor-specific and misleading configuration that makes it confusing to set up and not vendor-neutral:

- **Dead dependency**: `@asgardeo/react` is installed but never imported — the portal actually uses the standards-based `react-oidc-context` / `oidc-client-ts` stack.
- **Hardcoded IdP endpoints**: `src/main.tsx` manually defined Asgardeo-specific `/oauth2/*` and `/oidc/*` endpoints instead of relying on standard OIDC discovery.
- **Misleading env config**: `.env.template` / `.env.example` documented `VITE_*` variables that the app never actually reads. The app is configured purely at runtime via `public/config.js` (`window.configs`); the only build-time vars (`VITE_PORT`, `VITE_BASE_PATH`) were redundant with Vite defaults.
- **Confusing runtime keys**: `config.js` keys used a meaningless `VITE_` prefix (Vite never reads them — they're plain runtime values) with inconsistent casing, and ambiguous names (`apiUrl` vs `VITE_BASE_URL`).
- **Duplicated types**: the `Window.configs` interface was declared separately in both `main.tsx` and `App.tsx`, free to drift.
- **Stale docs**: `README.md` referenced non-existent env vars, the wrong dev port, a wrong URL query param, and a Docker section for a Dockerfile flow that doesn't match reality.
- **Cross-portal inconsistency**: `admin-portal` already migrated its IdP env vars to `VITE_IDP_*`, while consent-portal still used flat `VITE_CLIENT_ID` / `VITE_BASE_URL` / `VITE_SCOPE`.

## Suggested Improvement

Make the consent-portal configuration vendor-neutral, self-consistent, and correctly documented:

- Remove the unused `@asgardeo/react` dependency.
- Replace the hardcoded `/oauth2/*` endpoints in `main.tsx` with standard **OIDC discovery** (`{idpBaseUrl}/.well-known/openid-configuration`).
- Delete the misleading `.env.template` / `.env.example` and the redundant `VITE_PORT` / `VITE_BASE_PATH` build config (rely on Vite defaults).
- Rename the runtime `config.js` keys to clear, consistent, `idp`-prefixed camelCase: `consentEngineUrl`, `idpClientId`, `idpBaseUrl`, `idpScope`, `idpSignInRedirectUrl`, `idpSignOutRedirectUrl`.
- Add a tracked `public/config.example.js` template (the real `config.js` is gitignored).
- Dedupe the `Window.configs` interface into a shared `src/types/config.ts` (`AppConfig`).
- Update `entrypoint.sh` and `setup-portals.sh` to emit the new keys, and align the input container env vars to admin-portal's `VITE_IDP_*` convention.
- Refresh `README.md` to document the runtime-config setup accurately.

**Note (breaking deployment change):** the input container env vars for the IdP are renamed to `VITE_IDP_CLIENT_ID` / `VITE_IDP_BASE_URL` / `VITE_IDP_SCOPE`. Because `entrypoint.sh` uses `${VAR:-}` fallbacks, any deployment still setting the old names will silently produce an empty `config.js` (broken OIDC). Deployment configs must be updated in lockstep.

## Version

commit 96b5a16c (branch `refactor/consent-portal-vendor-neutral`)

## Additional Context

Follow-up: `member-portal` still uses the old flat `config.js` key names (`apiUrl`, `VITE_CLIENT_ID`, ...) and `admin-portal` uses `VITE_IDP_*` as config keys — the three portals should be unified in a later change.

### PR DESCRIPTION
**Pull request by mushrafmim**
## Summary

Makes the **consent-portal** configuration vendor-neutral, self-consistent, and correctly documented. The portal previously carried an unused Asgardeo SDK, hardcoded Asgardeo-specific OIDC endpoints, misleading env config, and inconsistent runtime config keys. This PR removes the vendor coupling, switches to standard OIDC discovery, and aligns naming with the already-migrated `admin-portal`.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Documentation update
- [x] Refactoring (no functional changes)
- [ ] Performance improvement
- [ ] Other (please describe):

## Changes Made

- **Removed unused dependency**: dropped `@asgardeo/react` (the portal uses the standards-based `react-oidc-context` / `oidc-client-ts` stack).
- **Standard OIDC discovery**: `src/main.tsx` no longer hardcodes Asgardeo-specific `/oauth2/*` and `/oidc/*` endpoints; they are now resolved from `{idpBaseUrl}/.well-known/openid-configuration`.
- **Removed misleading env config**: deleted `.env.template` / `.env.example` and the redundant `VITE_PORT` / `VITE_BASE_PATH` build config (Vite defaults cover both). The app is configured at runtime via `public/config.js` (`window.configs`).
- **Renamed runtime config keys** to clear, `idp`-prefixed camelCase: `consentEngineUrl`, `idpClientId`, `idpBaseUrl`, `idpScope`, `idpSignInRedirectUrl`, `idpSignOutRedirectUrl`.
- **Added `public/config.example.js`** as a tracked template (the real `config.js` is gitignored).
- **Deduped types**: consolidated the duplicated `Window.configs` interface (previously in both `main.tsx` and `App.tsx`) into a shared `src/types/config.ts` (`AppConfig`).
- **Updated deployment tooling**: `entrypoint.sh` and `setup-portals.sh` now emit the new config keys, and the container input env vars were renamed to drop the misleading `VITE_` prefix (they are consumed by `entrypoint.sh` at runtime, not by Vite at build time): `CONSENT_ENGINE_URL`, `IDP_CLIENT_ID`, `IDP_BASE_URL`, `IDP_SCOPE`, `IDP_SIGN_IN_REDIRECT_URL`, `IDP_SIGN_OUT_REDIRECT_URL`.
- **Cleaned up `Dockerfile`**: removed the dead build-stage `VITE_PORT` / `VITE_BASE_PATH` ARG/ENV block.
- **Refreshed `README.md`** to document the runtime-config setup accurately (correct port, setup step, config key table).

## Testing

- [x] I have tested this change locally
- [ ] I have added unit tests for new functionality
- [ ] I have tested edge cases
- [ ] All existing tests pass

Verified `npm run build` (`tsc -b && vite build`) passes cleanly; the only build log warning (`config.js can't be bundled`) is pre-existing and expected for the runtime-config pattern. Shell scripts (`entrypoint.sh`, `setup-portals.sh`) pass `sh -n` / `bash -n` syntax checks. No unit tests were added — the change is config/wiring with no new logic to unit test.

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #464

## Screenshots/Demo

N/A — configuration and tooling changes only.

## Additional Notes

Follow-up (out of scope): `member-portal` still uses the old flat `config.js` key names (`apiUrl`, `VITE_CLIENT_ID`, ...), and `admin-portal` uses `VITE_IDP_*` as its config keys. The three portals should be unified in a later change.

## Deployment Notes

**Breaking deployment change.** All container input env vars are renamed to drop the `VITE_` prefix (it was misleading — these are read by `entrypoint.sh` at runtime, not by Vite at build time):

- `VITE_API_URL` → `CONSENT_ENGINE_URL`
- `VITE_CLIENT_ID` → `IDP_CLIENT_ID`
- `VITE_BASE_URL` → `IDP_BASE_URL`
- `VITE_SCOPE` → `IDP_SCOPE`
- `VITE_SIGN_IN_REDIRECT_URL` → `IDP_SIGN_IN_REDIRECT_URL`
- `VITE_SIGN_OUT_REDIRECT_URL` → `IDP_SIGN_OUT_REDIRECT_URL`

Because `entrypoint.sh` uses `${VAR:-}` fallbacks, any deployment still setting the old names will **silently generate an empty `config.js`** (broken OIDC discovery, no error). Deployment configuration (CI / k8s / compose, which live outside this repo) must be updated to the new names in lockstep with this merge.

### CODE REVIEW

File: portals/consent-portal/src/main.tsx [hunk 1]
```diff
@@ -4,39 +4,15 @@ import { AuthProvider, type AuthProviderProps } from "react-oidc-context";
...
-    issuer: `${window?.configs?.VITE_BASE_URL}/oauth2/token`,
-  },
+  authority: window?.configs?.idpBaseUrl, // Endpoints are resolved via OIDC discovery (.well-known/openid-configuration)
+  client_id: window?.configs?.idpClientId,
+  redirect_uri: window?.configs?.idpSignInRedirectUrl,
+  post_logout_redirect_uri: window?.configs?.idpSignOutRedirectUrl,
+  scope: window?.configs?.idpScope || 'openid profile email',
   onSigninCallback: () => {
     // Remove query params (code, state) after successful login
     window.history.replaceState({}, document.title, window.location.pathname);
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portals/consent-portal/src/main.tsx]: Addressed.

File: portals/setup-portals.sh [hunk 2]
```diff
@@ -103,7 +103,7 @@ verify_portal() {
             required_vars=("VITE_API_URL" "VITE_LOGS_URL" "VITE_IDP_CLIENT_ID" "VITE_IDP_BASE_URL" "VITE_IDP_SCOPE" "VITE_IDP_ADMIN_ROLE")
             ;;
         "consent-portal")
-            required_vars=("apiUrl" "VITE_CLIENT_ID" "VITE_BASE_URL" "VITE_SCOPE")
+            required_vars=("consentEngineUrl" "idpClientId" "idpBaseUrl" "idpScope")
```
[the remark this answers is not in the document]
Comment (mushrafmim) [reply portals/setup-portals.sh]: Addressed.

File: portals/consent-portal/Dockerfile [hunk 3]
```diff
@@ -39,9 +30,9 @@ RUN chmod +x /entrypoint.sh
 
 # Set default environment variables (can be overridden at runtime)
 ENV VITE_API_URL=""
-ENV VITE_CLIENT_ID=""
-ENV VITE_BASE_URL=""
-ENV VITE_SCOPE=""
+ENV VITE_IDP_CLIENT_ID=""
```
Comment (ginaxu1) [inline portals/consent-portal/Dockerfile]: for consistency, rename Docker ENV vars to match the runtime config key names, then update `entrypoint.sh` accordingly

suggested rename: 
```
ENV CONSENT_ENGINE_URL=""
ENV IDP_CLIENT_ID=""
ENV IDP_BASE_URL=""
ENV IDP_SCOPE=""
ENV IDP_SIGN_IN_REDIRECT_URL=""
ENV IDP_SIGN_OUT_REDIRECT_URL=""
```

**Review by ginaxu1 — CHANGES_REQUESTED** [review]

File: portals/consent-portal/Dockerfile (hunk 3 above)
Comment (mushrafmim) [reply portals/consent-portal/Dockerfile]: Addressed.

**Review by ginaxu1 — APPROVED** [review]
Lgtm
